### DISCUSSION TITLE
Problem when using optitrack: NatNetSDK Error 3

### DISCUSSION THREAD

**Discussion by Carmol1** [USC-ACTLab/crazyswarm#650] [Q&A]
Hello.

The reason by which I open this issue it is due to the following problem:
![Captura de pantalla de 2022-05-17 18-34-42](https://user-images.githubusercontent.com/104467412/169142821-48af4990-d6e6-4a82-b87d-fb6da6573d0b.png)
I am using the motive software version 1.9, by which I have set up hover_swarm.launch as I show to below:

 
      motion_capture_type: "optitrack_closed_source" # one of none,vicon,optitrack,optitrack_closed_source,qualisys,vrpn
      object_tracking_type: "motionCapture" # one of motionCapture,libobjecttracker
      send_position_only: False # set to False to send position+orientation; set to True to send position only
      motion_capture_host_name: "192.168.0.102" pc with ubuntu 20.04
      motion_capture_interface_ip: "192.168.0.100" # optional for optitrack with multiple interfaces pc with windows
      save_point_clouds: "/dev/null" # set to a valid path to log mocap point cloud binary file.
      print_latency: False
      write_csvs: False
      force_no_cache: False
      enable_parameters: True
      enable_logging: False
      enable_logging_pose: True

On the other hand, in crazyflies.yaml I heve put

crazyflies:
- channel: 80
  id: 1
  initialPosition: [1.5, 1.5, 0.0]
  type: medium

I have looked if other users have had the same problem, but I have not found similar results.

I'd appreciate their help.

**Comment by whoenig** [comment]
Have you tried running the NatNetSDK outside from the Crazyswarm? If so, does this work for you? The source code and examples can be downloaded from the vendor, or found at https://github.com/whoenig/NatNetSDKCrossplatform.

**Reply by Carmol1** [reply]
Hello. 

I appreciate your help, I have already been able to solve the problem by changing motion_capture_host_name: "192.168.0.102" to "192.168.0.100" ip from the pc with motive 1.9, on the other hand, I have already run NatNetSDK on a PC with ubuntu 20.04 and I have data, however when I run it on a virtual box with ubuntu 20.04 and running ./packetClient 192.168.0.100 I get:

NatNetVersion: 0.0.0.0
Server Version: 2.9.0.0
set_option: no such device

For this reason I would like to know if it has a solution or it is not compatible to be used in a virtual box?

Greetings.

**Reply by whoenig** [reply]
This depends on the network settings for your virtual box. You might get more help in a forum around Virtual Box. In general, we don't recommend using virtual machines due to the added latency.

**Reply by Carmol1** [reply]
Thanks for the information, I will try not to use the virtual box, just to mention, when using vrpn I was able to use hover_swarm.launch, so my problem is solved.

Regards.
